Tests/scripts/common.py

```diff
@@ -217,3 +219,106 @@ def replace_escape_characters(sentence: str, replace_with: str = " ") -> str:
+    """
+    Get all pipelines and commits on the master branch in the last X hours,
+    pipelines are filtered to only include successful and failed pipelines.
+    Args:
+        gitlab_url - the url of the gitlab instance
+        gitlab_access_token - the access token to use to authenticate with gitlab
+        project_id - the id of the project to query
+        look_back_hours - the number of hours to look back for commits and pipeline
+    Return:
+        a list of gitlab pipelines and a list of gitlab commits in ascending order
+    """
+    gl = gitlab.Gitlab(gitlab_url, private_token=gitlab_access_token)
+    project = gl.projects.get(project_id)
+
+    # Calculate the timestamp for look_back_hours ago
+    time_threshold = (
+        datetime.utcnow() - timedelta(hours=look_back_hours)).isoformat()
+
+    commits = project.commits.list(all=True, since=time_threshold, order_by='updated_at', sort='asc')
+    pipelines = project.pipelines.list(all=True, updated_after=time_threshold, ref='master',
+                                       source='push', order_by='updated_at', sort='asc')
+
+    # Filter out pipelines that are not done
+    filtered_pipelines = [
+        pipeline for pipeline in pipelines if pipeline.status in ('success', 'failed')]
+
+    return filtered_pipelines, commits
+
+
+def shame(commit):
+    """
+    Returns the name, email, and PR link for the author of the provided commit.
+
+    Args:
+        commit: The Gitlab commit object containing author info.
+
+    Returns:
+        name: The name of the commit author.
+        email: The email of the commit author.
+        pr: The GitHub PR link for the Gitlab commit.
+    """
+    name = commit.author_name
+    email = commit.author_email
+    pr_number = commit.title.split()[-1][2:-1]
+    pr = f"https://github.com/demisto/content/pull/{pr_number}"
+    return name, email, pr
+
+
+def are_pipelines_in_order_as_commits(commits, current_pipeline_sha, previous_pipeline_sha):
+    """
+    This function checks if the commit that triggered the current pipeline was pushed
+    after the commit that triggered the the previous pipeline,
+    to avoid rare condition that pipelines are not in the same order as their commits.
+    Args:
+        commits: list of gitlab commits
+        current_pipeline_sha: string
+        previous_pipeline_sha: string
+
+    Returns:
+        boolean , the commit that triggered the current pipeline
+    """
+    current_pipeline_commit_timestamp = None
+    previous_pipeline_commit_timestamp = None
+    the_triggering_commit = None
+    for commit in commits:
+        if commit.id == previous_pipeline_sha:
+            previous_pipeline_commit_timestamp = parser.parse(commit.created_at)
+        if commit.id == current_pipeline_sha:
+            current_pipeline_commit_timestamp = parser.parse(commit.created_at)
+            the_triggering_commit = commit
```

Comment:
This is only basic CR per your request but this can and should be written cleaner in prod.

Reply:
agree